### PR TITLE
Fix quotation for timestamps

### PR DESCRIPTION
**Pull request by shuheiktgw**
Closes https://github.com/goccy/go-yaml/issues/430

Before submitting your PR, please confirm the following.

- [x] Describe the purpose for which you created this PR.  
- [x] Create test code that corresponds to the modification

### CODE REVIEW

File: token/token.go [hunk 1]
```diff
@@ -600,20 +601,23 @@ func ToNumber(value string) *NumberValue {
 	}
 }
 
-func looksLikeTimeValue(value string) bool {
```
Comment (shuheiktgw) [inline token/token.go:603]: I've looked at [the original PR](https://github.com/goccy/go-yaml/pull/158), yet it was not clear to me why we needed to treat time values differently, so I removed the logic at all, but I'm entirely sure if that was OK or not...

File: token/token.go [hunk 2]
```diff
@@ -600,20 +601,26 @@ func ToNumber(value string) *NumberValue {
...
-		case ':', '1', '2', '3', '4', '5', '6', '7', '8', '9':
-			continue
-		case '0':
-			if i == 0 {
-				return false
-			}
-			continue
+// This is a subset of the formats permitted by the regular expression
+// defined at http://yaml.org/type/timestamp.html. Note that time.Parse
+// cannot handle: "2001-12-14 21:59:43.10 -5" from the examples.
```
Comment (goccy) [inline token/token.go:606]: Why didn't you provide a layout to handle this case ?

Comment (shuheiktgw) [reply token/token.go:606]: Yes I've actually tried supporting the layout, but Go does not seem to support the "-timezone" representation...

https://go.dev/play/p/a5UvI5cUlJy

Comment (goccy) [reply token/token.go:606]: OK ! I understand !

**Comment by goccy** [comment]
LGTM 👍

**Pull request merged**